proof/crefine/RISCV64/SchedContext_C.thy

```diff
@@ -584,11 +584,251 @@ lemma schedContext_updateConsumed_ccorres:
+                 apply (clarsimp simp: to_bool_def)
+                apply (ctac (no_vcg) add: refill_update_ccorres)
+               apply (ctac (no_vcg) add: refill_new_ccorres)
+              apply wpsimp
+             (* target sc is not associated with a thread, so we perform refill_new *)
+             apply ccorres_rewrite
+             apply (rule ccorres_cond_false)
+             apply (ctac add: refill_new_ccorres)
+            apply vcg
+           (* target sc is not active, so we perform refill_new *)
+           apply clarsimp
+           apply ccorres_rewrite
+           apply (rule ccorres_cond_seq)
+           apply (rule ccorres_cond_false)
+           apply ccorres_rewrite
+           apply (rule ccorres_cond_seq)
+           apply (rule ccorres_cond_false)
+           apply ccorres_rewrite
+           apply (rule ccorres_cond_false)
+           apply (ctac (no_vcg) add: refill_new_ccorres)
+          apply wpsimp
+         apply ceqv
+        apply (rule_tac r'=dc and xf'=xfdc in ccorres_split_nothrow_novcg)
+            apply (clarsimp simp: when_def)
+            apply (rule_tac Q="obj_at' (\<lambda>sc'. scTCB sc' = scTCB sc) scPtr and valid_objs'"
+                         in ccorres_cond_both'[where Q'=\<top>])
+              apply normalise_obj_at'
+              apply (frule (1) sc_ko_at_valid_objs_valid_sc')
+              apply (frule (1) obj_at_cslift_sc)
+              apply (clarsimp simp: option_to_ctcb_ptr_def valid_sched_context'_def)
+              apply (case_tac "scTCB sc"; clarsimp)
+              apply (fastforce dest: tcb_at_not_NULL)
+             apply (rule ccorres_rhs_assoc)
+             apply (ctac add: schedContext_resume_ccorres)
+               apply (rule ccorres_rhs_assoc)+
+               apply (ctac add: isRunnable_ccorres, rename_tac runnable)
+                 apply (rule ccorres_pre_getCurThread, rename_tac ctPtr)
+                 apply (rule ccorres_cond_seq)
+                 apply ccorres_rewrite
+                 apply clarsimp
+                 apply (drule Some_to_the)
+                 apply (rule_tac P="the (scTCB sc) = ctPtr" in ccorres_cases)
+                  (* the target sc is associated with the current thread,
+                     so we perform rescheduleRequired *)
+                  apply clarsimp
+                  apply (rule ccorres_cond_false)
+                  apply ccorres_rewrite
+                  apply (rule ccorres_cond_true)
+                  apply (ctac add: rescheduleRequired_ccorres)
+                 apply (clarsimp simp: when_def to_bool_def)
+                 apply (rule_tac Q="\<lambda>s. ctPtr = ksCurThread s" in ccorres_cond_both'[where Q'=\<top>])
+                   apply (clarsimp dest!: rf_sr_ksCurThread simp: option_to_ctcb_ptr_def)
+                  (* the target sc is not associated with the current thread, but is associated
+                     with a runnable thread, so we perform possibleSwitchTo *)
```

Comment:
comment on commentary: this "so we perform" is both novel and kind of long-winded, could I suggest that we go back to calling or invoking functions? i.e. "the target is [...], call possibleSwitchTo"
(same for other occurrences)

Reply:
to be fair, in the context of seL4, we do "perform" invocations, but we shouldn't extend that to everything